src/audio/tdfb/tdfb.c

```diff
@@ -673,6 +668,9 @@ static void tdfb_process(struct comp_dev *dev, struct comp_buffer *source,
 	/* calc new free and available */
 	comp_update_buffer_consume(source, source_bytes);
 	comp_update_buffer_produce(sink, sink_bytes);
+
+	/* Update sound direction estimate */
+	tdfb_direction_estimate(cd, frames, source->stream.channels);
```

Comment:
The control events are not yet generated. Need to add filtering and hysteresis to avoid user space to be woken up too often.

Reply:
done